Project/Source/Utils/Trail.cpp

```diff
@@ -35,12 +35,20 @@ void Trail::Init() {
 	EditTextureCoords();
 }
 
+bool Trail::CalculateDistance(float3 A, float3 B) {
+	if (Abs(B.x - A.x) > vertexDistance) return true;
+	if (Abs(B.y - A.y) > vertexDistance) return true;
+	if (Abs(B.z - A.z) > vertexDistance) return true;
+	return false;
+}
 void Trail::Update(float3 mPosition) {
 	if (!isRendering) return;
+	float3 currentDirection = (previousPosition - currentPosition).Normalized();
```

Comment:
direction is  (currentPosition -previousPosition). Care then with the Perpendicular() result